### ISSUE TITLE
使用 pnpm 安装 vue create 会 报错

### ISSUE THREAD

**Issue by king-open** [vuejs/vue-cli#7284]
### Version
5.0.8

### Environment info
```
macos12.5  pnpm7.9.0
```

### Steps to reproduce
`vue create back`

选择 `pnpm`  报错

### What is expected?
正常能够使用pnpm构建项目依赖

### What is actually happening?
安装使用pnpm 出现以下错误
`command failed: pnpm install --reporter silent --shamefully-hoist --registry=https://registry.npmmirror.com/`

**Comment by king-open** [comment]
Follow the tips my friend gave me
```
mkdir back && cd back && pnpm init && vue create . 
```
still wrong

**Comment by king-open** [comment]
I have tried to use create-vue in slowly getting used to

**Comment by lvqq** [comment]
You can add `strict-peer-dependencies=false` in `.npmrc` file and it may work

**Comment by king-open** [comment]
nope, same error

**Comment by lvqq** [comment]
Yeah, I just found it that the error interrupted the process of installing, and the next process of generating files wouldn't execute. So the method won't work, maybe another solution is to degrade pnpm to 6.x because the installing is build-in.

**Comment by linghengqian** [comment]
You can start a new line in `.npmrc` and add `auto-install-peers=true`.  This works for pnpm v6 and pnpm v7.  👀

**Comment by king-open** [comment]
i try first

**Comment by king-open** [comment]
Wow, lowering pnpm to 6 works fine . amazing

**Comment by king-open** [comment]
thanks man

**Comment by lvqq** [comment]
> You can start a new line in `.npmrc` and add `auto-install-peers=true`.  This works for pnpm v6 and pnpm v7.  👀

That won't work because .npmrc file will be overwrite by cli🤣

**Comment by king-open** [comment]
Yeah

**Comment by linghengqian** [comment]
> > You can start a new line in `.npmrc` and add `auto-install-peers=true`.  This works for pnpm v6 and pnpm v7.  👀
> 
> That won't work because .npmrc file will be overwrite by cli🤣

- There are two ways.  
- One is to submit a PR with additional processing `auto-install-peers=true`.  I'm still waiting for a friend who has permission to merge the PR to review and compare https://github.com/vuejs/vue-cli/pull/7265.  
- The other is to switch nodejs corepack to pnpm v6 to initialize the project, and then switch nodejs corepack to pnpm v7 to execute `pnpm install`, which also requires manually adding `auto-install-peers=true`.

**Comment by king-open** [comment]
So if I add that sentence to .npmrc first, it's fine

**Comment by linghengqian** [comment]
This issue was fixed in #7265 and should be closed.

**Comment by king-open** [comment]
ok thanks

**Comment by Wang-zhetao** [comment]
Why doesn't it work anymore

**Comment by zkl00** [comment]
pnpm i 提示这个错误
 no such file or directory, stat 'Users\admin\Desktop\demo\dome_vue3\node_modules\.pnpm\jsdom@21.1.0\node_modules\decimal.js'

这是什么原因
